src/base/private/profile_p.cpp

```diff
@@ -105,6 +105,8 @@ Private::CustomProfile::CustomProfile(const QString &rootPath, const QString &co
     : Profile {configurationName}
     , m_rootDirectory {QDir(rootPath).absoluteFilePath(this->profileName())}
 {
+    if (!QDir().mkpath(m_rootDirectory.absolutePath()))
```

Comment:
Are you sure it's realty needed? Seems we create some subfolders inside it so it should be created as well.
https://github.com/Tester798/qBittorrent/blob/062bc35547a74b3f7e0ad0ab575420023fd9ef21/src/base/profile.cpp#L42

Reply:
Just tested and it shows that code at https://github.com/Tester798/qBittorrent/blob/062bc35547a74b3f7e0ad0ab575420023fd9ef21/src/base/profile.cpp#L42 is executed at line https://github.com/Tester798/qBittorrent/blob/062bc35547a74b3f7e0ad0ab575420023fd9ef21/src/base/profile.cpp#L58 but we need existing directory at line https://github.com/Tester798/qBittorrent/blob/062bc35547a74b3f7e0ad0ab575420023fd9ef21/src/base/profile.cpp#L55 already.

Reply:
Well, seems like design issue... we need to move these "ensure" calls to `Private::Profile` constructor (and move entire method definition into that class).

Reply:
Should I try to do it in this pull request?
Tried to move "ensureDirectoryExists" calls to Private::Profile in commit 96687fb but not sure if I did it correctly. Tested on windows and looks like it's working fine.

Reply:
>Should I try to do it in this pull request?

IMO, Yes. Why do you should make workaround instead of fixing the problem?..